src/components/SecretsModal/Annotations.js

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { TextInput } from 'carbon-components-react';
+import Add from '@carbon/icons-react/lib/add--alt/24';
+import Remove from '@carbon/icons-react/lib/misuse/24';
+import './SecretsModal.scss';
+
+const Annotations = props => {
+  const {
+    annotations,
+    handleChange,
+    handleAdd,
+    handleRemove,
+    invalidFields,
+    disabled
+  } = props;
+
+  const annotationFields = [];
+  for (let i = 0; i < annotations.length; i += 2) {
+    annotationFields.push(
```

Comment:
using map in this case is kinda difficult since I'm iterating every two elements in the array instead of one because at each iteration im creating one row of annotation composed of 2 text inputs which with the `i` im able to access the element needed for the respective input.

Reply:
Do we control this data structure? If so I'd recommend changing it so we have an array of objects instead, each object could have the label + value required

Reply:
yeah I have it to be stored as an array of strings but I can change it to that.